var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -43,11 +44,12 @@ class PyH5py(PythonPackage):
 
     # Build dependencies
     depends_on("py-cython@0.23:", type="build", when="@:2")
```

Comment:
```suggestion
    depends_on("py-cython@0.23:0", type="build", when="@:2")
```
Let's prevent this for ancient versions too.